src/shared/mount-setup.c

```diff
@@ -148,13 +149,34 @@ bool mount_point_ignore(const char *path) {
         return false;
 }
 
+static bool is_old_kernel(void) {
+    bool old = false;
+    struct utsname uts;
+    int r = 0;
+
+    r = uname(&uts);
+    if (r < 0)
+            return old;
+
+    if (strverscmp(uts.release, "5.7") < 0) {
+            log_debug("Pre v5.7 kernel detected [v%s] - skipping memory_recursiveprot", uts.release);
+            old = true;
+    }
+
+    return old;
+}
+
 static int mount_one(const MountPoint *p, bool relabel) {
         int r, priority;
 
         assert(p);
 
         priority = (p->mode & MNT_FATAL) ? LOG_ERR : LOG_DEBUG;
 
+        if (streq(p->what, "cgroup2") && p->options && streq(p->options, "nsdelegate,memory_recursiveprot") && is_old_kernel()) {
+            return 0;
+        }
```

Comment:
something like this might be mergable, but please use `mount_option_supported()` to ask the kernel directly if the mount option is supported, instead of binding this to a kernel version. Distros typically backport things, and sometimes quite heavily, hence we generally never to version checking, always feature checking.

Also, please follow coding style, i.e. 8 space indentation, and no {} on single-line if blocks:

https://systemd.io/CODING_STYLE/

Reply:
also, pease implementation this as a `condition_fn` for the relevant fsype entry, i.e. the 6th field in the `MountPoint` structure, see above.

Reply:
> also, pease implementation this as a `condition_fn` for the relevant fsype entry, i.e. the 6th field in the `MountPoint` structure, see above.

The `condition_fn` of this fstype entry has already been set with  'cg_is_unified_wanted' ,so can this check replace it?

Reply:
replace it with a new function, that first calls cg_is_unified_wanted(), and then does your additional check.